packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -386,12 +386,10 @@ export const DatePickerMixin = (subclass) =>
           }
         }
 
-        if (this.inputElement.value === '' && this.__dispatchChange) {
-          this.validate();
+        this.validate();
```

Comment:
Moved `validate()` outside of the `if` because it's called regardless of the condition.